drivers/hmis/app/models/hmis/ce/opportunity.rb

```diff
@@ -17,10 +17,10 @@ class Opportunity < GrdaWarehouseBase
                primary_key: 'identifier',
                class_name: 'Hmis::WorkflowDefinition::Template'
 
-    has_many :referrals, class_name: 'Hmis::Ce::Referral', dependent: :restrict_with_exception
+    has_many :referrals, class_name: 'Hmis::Ce::Referral' # use default dependent behavior, since opportunities are soft-deleted
```

Comment:
I think this comment is a little confusing. For my own understanding, the default behavior is: when an Opportunity is destroyed, the associated referrals continue to reference the soft-deleted opportunity record (in `ce_referrals` `opportunity_id` column)

Maybe this is clearer?

```suggestion
    has_many :referrals, class_name: 'Hmis::Ce::Referral' # Referrals continue to reference the opportunity even after it is soft-deleted, as no dependent behavior is specified.
```

Reply:
@martha This adds some interesting complexity to the relationships. is this change proposed because of the following scenario:
- unit is marked available (opportunity created)
- referral is created and declined
- unit is marked unavailable (opportunity destroyed)
?



When a referral is _accepted_, [we close!](https://github.com/greenriver/hmis-warehouse/blob/66b3d16c9c539e38265a6a133e13a793bf13b71e/drivers/hmis/app/models/hmis/ce/referral_message_handler.rb#L73) the opportunity.

However when a unit is marked as no longer available, we destroy it. I wonder if it would make more sense to call `close!` here instead of destroy: https://github.com/greenriver/hmis-warehouse/blob/66b3d16c9c539e38265a6a133e13a793bf13b71e/drivers/hmis/app/graphql/mutations/ce/mark_units_unavailable.rb#L33 ? I don't think we are deriving any value/meaning from the distinction that an opportunity was "destroyed" vs being "closed".

That approach would allow us to keep the `dependent: :restrict_with_exception` restriction, ensuring that any Referral records reference a non-deleted Opportunity. That feels less complicated to me. What do you think?

Reply:
That sounds reasonable to me too! Thanks!